src/bin/main.rs

```diff
@@ -122,14 +121,16 @@ fn validate(args: &ArgMatches) -> HowserResult<ValidationReport> {
     }
 }
 
-fn check(args: &ArgMatches) -> HowserResult<ValidationReport> {
+fn check(args: &ArgMatches) -> HowserResult<ValidationProblems> {
     if let Some(filename) = args.value_of("prescription") {
         let filename = String::from(filename);
         let rx_root = parse_document(&get_file_contents(&filename)?);
         let document = Document::new(&rx_root, Some(filename));
-        let rx = document.into_prescription()?;
-        let checker = SpecChecker::new(SpecRule::standard_set());
-        Ok(checker.check(&rx)?)
+        match document.into_prescription() {
+            Err(HowserError::PrescriptionError(warning)) => Ok(Some(vec![Box::new(warning)])),
```

Comment:
I'm going to update it to
`type ValidationProblem = Box<Reportable>`
and have these things pass around `Option<ValidationProblem>`.
Since for now, I am only dealing with one reportable problem at a time. Does that sound reasonable?

Reply:
Yeah, that sounds like an improvement for sure.